lib/internal/crypto/random.js

```diff
@@ -281,10 +282,86 @@ function getRandomValues(data) {
   return data;
 }
 
+// Implements an RFC 4122 version 4 random UUID.
+// To improve performance, random data is generated in batches
+// large enough to cover kBatchSize UUID's at a time. The uuidData
+// and uuid buffers are reused for each. Each call to randomUUID()
+// consumes 16 bytes from the buffer.
+
+const kHexDigits = [
+  48, 49, 50, 51, 52, 53, 54, 55,
+  56, 57, 97, 98, 99, 100, 101, 102
+];
+
+const kBatchSize = 128;
+let uuidData;
+let uuid;
+let uuidBatch = 0;
+
+function randomUUID() {
+  if (uuidData === undefined) {
+    uuidData = new Uint8Array(16 * kBatchSize);
+    uuid = Buffer.alloc(36, '-');
+    uuid[14] = 52; // '4'
```

Comment:
I think this non-trivial line that includes magic numbers warrants a more descriptive comment.